source/geometry_model/box.cc

```diff
@@ -81,10 +84,10 @@ namespace aspect
     {
       // Here we provide GridTools with the function to displace vertices
       // in the vertical direction by an amount specified by the initial topography model
-      GridTools::transform(std::bind(&Box<dim>::add_topography,
-                                     this,
-                                     std::placeholders::_1),
-                           grid);
+      GridTools::transform(
+        [&](const Point<dim> &p) -> Point<dim>
+      {return this->add_topography(p);},
```

Comment:
above you have newlines between the braces and the lambda body, do we have a convention for this yet?